test/Npgsql.Tests/CommandTests.cs

```diff
@@ -288,6 +288,25 @@ public async Task Cancel()
             await cancelTask;
         }
 
+        [Test]
+        public async Task CancelAsyncImmediately()
+        {
+            if (IsMultiplexing)
+                return; // Multiplexing, cancellation
+
+            using var cts = new CancellationTokenSource();
+            cts.Cancel();
+
+            await using var conn = await OpenConnectionAsync();
+            using var cmd = conn.CreateCommand();
+            cmd.CommandText = "SELECT 1";
+
+            Assert.ThrowsAsync<OperationCanceledException>(async () => await cmd.ExecuteScalarAsync(cts.Token));
```

Comment:
If we want a really tight test, we can assert that ExecuteScalarAsync *synchronously* returns a cancelled task (without doing any await). This would make sure nothing has been sent to the database etc.